extensions/wikia/CreateNewWiki/CreateNewWikiController.class.php

```diff
@@ -274,55 +274,93 @@ public function CreateWiki() {
-				]
-			);
-			$this->response->setCode( 500 );
-			$this->response->setVal( self::STATUS_FIELD, self::STATUS_BACKEND_ERROR );
-			$this->response->setVal( self::STATUS_MSG_FIELD, wfMessage( 'cnw-error-general' )->parse() );
-			$this->response->setVal( self::STATUS_HEADER_FIELD, wfMessage( 'cnw-error-general-heading' )->escaped() );
-			$this->response->setVal( self::ERROR_CLASS_FIELD, get_class( $ex ) );
-			$this->response->setVal( self::ERROR_CODE_FIELD, $ex->getCode() );
-			$this->response->setVal( self::ERROR_MESSAGE_FIELD, $ex->getMessage() );
-			wfProfileOut( __METHOD__);
-			return;
-		}
+		$task->call( 'create', $params['wName'], $params['wDomain'], $params['wLanguage'], $params['wVertical'], $categories, $allAges );
+		$task_id = $task->setQueue( Wikia\Tasks\Queues\PriorityQueue::NAME )->queue();
 
-		$cityId = $createWiki->getCityId();
+		// return ID of the created task ID, front-end code will poll its status
+		// e.g {"task_id":"mw-04CF2876-2176-4036-96E5-B6378DB1AC8E"}
+		$this->response->setCode( 201 ); // HTTP 201 Created
+		$this->response->setVal( 'task_id', $task_id );
 
-		if ( isset($params['wAllAges']) && !empty( $params['wAllAges'] ) ) {
-			WikiFactory::setVarByName( self::WF_WDAC_REVIEW_FLAG_NAME, $cityId, true, __METHOD__ );
-		}
+		// return timestamp, so that front-end can send it back to the backend
+		// we'll compare it with the current timestamp in CheckStatus method
+		$this->response->setVal( 'timestamp', time() );
 
-		$this->response->setVal( self::STATUS_FIELD, self::STATUS_OK );
-		$this->response->setVal( self::SITE_NAME_FIELD, $createWiki->getSiteName() );
-		$this->response->setVal( self::CITY_ID_FIELD, $cityId );
-		$finishCreateTitle = GlobalTitle::newFromText( "FinishCreate", NS_SPECIAL, $cityId );
+		wfProfileOut(__METHOD__);
+	}
 
-		$fullURL = $finishCreateTitle->getFullURL( [ 'editToken' => $wgUser->getEditToken() ] );
-		$finishCreateUrl = empty( $wgDevelDomains ) ? $fullURL : str_replace( '.wikia.com', '.'.$wgDevelDomains[0], $fullURL );
-		$this->response->setVal( 'finishCreateUrl',  $finishCreateUrl );
+	/**
+	 * This AJAX entry point is used to poll for wiki creation status
+	 *
+	 * @see SUS-4383
+	 */
+	public function CheckStatus() {
+		global $wgDevelDomains;
+
+		$task_id = (string) $this->getRequest()->getVal('task_id');
+		$task_details = CreateWikiTask::getCreationLogEntry( $task_id );
+
+		// do not cache, we always want to hit the backend as the response can change anytime
+		$this->response->setCachePolicy( WikiaResponse::CACHE_PRIVATE );
+		$this->response->setCacheValidity( WikiaResponse::CACHE_DISABLED );
+		$this->response->setFormat( 'json' );
+
+		// compare the timestamp of when the creation started with the current one
+		$timestamp = (int) $this->getRequest()->getVal('timestamp');
+		$time_diff = time() - $timestamp;
+
+		// given task ID not found in the creation log (maybe not yet?)
+		if ( empty( $task_details ) ) {
+			if ( $time_diff > 60 ) {
```

Comment:
This check should also be performed in the task, so that we don't create a wiki if we already told the user there was an error.